docs/recipes.md

```diff
@@ -129,3 +129,56 @@ AsyncImage(
     },
 )
 ```
+
+## Transforming Requests
+
+You may need to transform HTTP requests used to fetch images. In this example, we'll use an [Interceptor](https://coil-kt.github.io/coil/api/coil-core/coil3.intercept/-interceptor) to append `width` & `height` query parameters to the request URL.
+
+```kotlin
+class FastlyCoilInterceptor : Interceptor {
+
+    override suspend fun intercept(chain: Chain): ImageResult {
+        val request = chain.request
+        val uri = request.uri
+
+        if (uri == null || uri.scheme !in setOf("https", "http")) {
+            // Ignore non-HTTP requests.
+            return chain.proceed()
+        }
+
+        val (width, height) = chain.size
+        return if (width is Pixels && height is Pixels) {
+            val transformedRequest = request.newBuilder()
+                .data(
+                    uri.buildUpon()
+                        .appendQueryParameter("width", "${width.px}")
+                        .appendQueryParameter("height", "${height.px}")
+                        .build()
+                )
+                .build()
+            return chain.withRequest(transformedRequest).proceed()
+        } else {
+            // Width & height aren't available, i.e. because of infinite constraints.
+            chain.proceed()
+        }
+    }
+
+    private val ImageRequest.uri: Uri?
```

Comment:
Is there a multi-platform friendly way to add query parameters to a `coil3.Uri`?

Reply:
Hmm there isn't at the moment, but we should consider adding one in a separate PR

Reply:
Filed https://github.com/coil-kt/coil/issues/3047